apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ReservedKeywords.kt

```diff
@@ -10,14 +10,17 @@ private val JAVA_RESERVED_WORDS = arrayOf(
     "transient", "try", "true", "void", "volatile", "while"
 )
 
-
-// Reference:
-// https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants:~:text=properties%20for%20obtaining%20its%20name%20and%20position
-private val KOTLIN_RESERVED_ENUM_VALUE_NAMES = arrayOf("name", "ordinal")
-
 fun String.escapeJavaReservedWord() = if (this in JAVA_RESERVED_WORDS) "${this}_" else this
 
 // Does nothing. KotlinPoet will add the backticks
 fun String.escapeKotlinReservedWord() = this
 
-fun String.escapeKotlinReservedEnumValueNames() = if (this in KOTLIN_RESERVED_ENUM_VALUE_NAMES) "${this}_" else this
+fun String.escapeKotlinReservedEnumValueNames() : String {
+  return when {
+    // https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants:~:text=properties%20for%20obtaining%20its%20name%20and%20position
+    this in arrayOf("name", "ordinal") -> "${this}_"
```

Comment:
Do you remember why we don't use backticks for these two as well? (we wouldn't want to change this now as it would be breaking, but I'm curious if we simply didn't think of it or a good reason).

Reply:
I wonder if you want to
```kotlin
"(?:name|ordinal)_*".toRegex().matches(this)
```
to prevent collisions if a schema defines both `name` and `name_`.

Reply:
Excellent point, much better 👍